de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/toolbar/ToolbarViewPresenterImpl.java

```diff
@@ -217,31 +220,29 @@ protected CreateFolderDialog getCreateFolderDialog(Folder selectedFolder) {
 
     @Override
     public void onCreateNcbiSraFolderStructure(final Folder selectedFolder) {
-        // FIXME Do not fire dialog from presenter. Do so from the view.
-        final CreateNcbiSraFolderStructureDialog dialog =
-                new CreateNcbiSraFolderStructureDialog(selectedFolder);
-        dialog.setHideOnButtonClick(false);
-        dialog.addOkButtonSelectHandler(new SelectHandler() {
-
+        asyncProviderWrapper.get(new AsyncCallback<CreateNcbiSraFolderStructureDialog>() {
             @Override
-            public void onSelect(SelectEvent event) {
-                parentPresenter.onCreateNcbiSraFolderStructure(selectedFolder,
-                                                               dialog.getProjectTxt(),
-                                                               dialog.getBiosampNum(),
-                                                               dialog.getLibNum());
-                dialog.hide();
-            }
-        });
+            public void onFailure(Throwable throwable) {
 
-        dialog.addCancelButtonSelectHandler(new SelectHandler() {
+            }
 
             @Override
-            public void onSelect(SelectEvent event) {
-                dialog.hide();
+            public void onSuccess(CreateNcbiSraFolderStructureDialog dialog) {
+                dialog.setHideOnButtonClick(false);
+                dialog.addOkButtonSelectHandler(event -> {
+                    if(dialog.validate()) {
+                        parentPresenter.onCreateNcbiSraFolderStructure(selectedFolder,
+                                                                       dialog.getProjectTxt(),
+                                                                       dialog.getBiosampNum(),
+                                                                       dialog.getLibNum());
+                        dialog.hide();
+                    }
+                });
+
+                dialog.addCancelButtonSelectHandler(event -> dialog.hide());
+                dialog.show(selectedFolder);
```

Comment:
Good refactor!  I think I did the same thing in my current open PR or upcoming PR 😄